pkg/antctl/raw/supportbundle/command.go

```diff
@@ -0,0 +1,486 @@
+	antrea "github.com/vmware-tanzu/antrea/pkg/client/clientset/versioned"
+)
+
+const (
+	barTmpl      pb.ProgressBarTemplate = `{{string . "prefix"}}{{bar . }} {{percent . }} {{rtime . "ETA %s"}}` // Example: 'Prefix[-->______] 20%'
+	requestRate                         = 50
+	requestBurst                        = 100
+)
+
+// Command is the support bundle command implementation.
+var Command *cobra.Command
+
+var option = &struct {
+	dir           string
+	labelSelector string
+}{}
+
+var remoteControllerLongDescription = strings.TrimSpace(`
+Generate support bundles for the cluster, which include: information about each Antrea agent, information about the Antrea controller and general information about the cluster.
+The controller support bundle will always be collected no matter whether the master Node is selected or not.
+`)
+
+var remoteControllerExample = strings.Trim(`
+  Generate support bundles of controller and agent on all Nodes and save them to current working dir
+  $ antctl supportbundle
+  Generate support bundles of specific Nodes filtered by name, with support for wildcard expressions
+  $ antctl supportbundles '*worker*'
+  Generate support bundles of specific Nodes that filtered by name and label selectors
+  $ antctl supportbundles '*worker*' -l kubernetes.io/os=linux
+  Generate support bundles of controller and agent on all Nodes and save them to specific dir
+  $ antctl supportbundle -d ~/Downloads
+`, "\n")
+
+func init() {
+	Command = &cobra.Command{
+		Use:   "supportbundle",
+		Short: "Generate support bundle",
+	}
+
+	if runtime.Mode == runtime.ModeAgent {
+		Command.RunE = agentRunE
+		Command.Long = "Generate the support bundle of current antrea agent."
+	} else if runtime.Mode == runtime.ModeController && runtime.InPod {
+		Command.RunE = controllerLocalRunE
+		Command.Long = "Generate the support bundle of current antrea controller."
+	} else if runtime.Mode == runtime.ModeController && !runtime.InPod {
+		Command.Args = cobra.MaximumNArgs(1)
+		Command.Use += " [nodeName]"
+		Command.Long = remoteControllerLongDescription
+		Command.Example = remoteControllerExample
+		cwd, _ := os.Getwd()
+		defaultOutputPath := filepath.Join(cwd, "support-bundles_"+time.Now().Format(time.RFC3339))
+		Command.Flags().StringVarP(&option.dir, "dir", "d", defaultOutputPath, "support bundles output dir, the path will be created if it not exists")
+		Command.Flags().StringVarP(&option.labelSelector, "label-selector", "l", "", "selector (label query) to filter on nodes, supports '=', '==', and '!='.(e.g. -l key1=value1,key2=value2)")
+		Command.RunE = controllerRemoteRunE
+	}
+}
+
+// TODO: enable secure connection.
+// TODO: generate kubeconfig in antrea agent for antctl in-Pod access.
+func setupKubeconfig(kubeconfig *rest.Config) {
+	kubeconfig.APIPath = "/apis"
+	kubeconfig.GroupVersion = &systemv1beta1.SchemeGroupVersion
+	kubeconfig.NegotiatedSerializer = scheme.Codecs.WithoutConversion()
+	kubeconfig.Insecure = true
```

Comment:
@tnqn @weiqiangt 
A bit of a last-minute question, is this only needed for the self-signed certificate case?

Reply:
I'm afraid not. There is no server verification function in Antctl for now.

Reply:
If antctl talks to kube-apiserver, it should use the K8s CA cert set in kubeconfig to verify kube-apiserver, and kube-apiserver will then start a new HTTPS request against antrea-controller and use CABundle in APIServices to verify it.
But support bundle command goes to antrea-controller and antrea-agent directly, we can probably read antrea-ca ConfigMap to verify antrea-controller, but we don't publish antrea-agent's self-signed cert yet, so no way to verify them for now.

Reply:
Thanks a lot for the clarification @tnqn.

@weiqiangt / @tnqn  do you think you could open an issue to track this? I don't know if you have a solution / long-term plan in mind already.

Reply:
Just created one #758.